### ISSUE TITLE
Solely studying one instance: vcpkg install error:in triplet x64-windows: Unable to find a valid Visual Studio instance Could not locate a complete Visual Studio instance

### ISSUE THREAD

**Issue by monty241** [microsoft/vcpkg#32506]
### Operating system

Windows Server 2019

### Compiler

cmake

### Steps to reproduce the behavior

Steps to execute: 
- Install Visual Studio Build Tools 2022.
- `vcpkg install protobuf` works.
- Then install Visual Studio Test Agent 2022 (2) (17.6.4) ("Agents for Visual Studio 2022" is name of download on https://visualstudio.microsoft.com/downloads/).
- Then include on Visual Studio Build Tools the individual component "MSVC v140 - VS 2015 C++ build tools (v14.00)"
- Create a local folder `c:\ws`.
- Make sure to set VCPKG_VISUAL_STUDIO_PATH in the command.com:
```
set VCPKG_VISUAL_STUDIO_PATH=c:\ws
```
- Then try to install another package, for instance grpc.
- Error now:

```
Computing installation plan...
The following packages will be built and installed:
    grpc:x86-windows -> 1.51.1
error: in triplet x86-windows: Unable to find a valid Visual Studio instance
    at "C:\ws"
The following Visual Studio instances were considered:
    C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\
The following paths were examined for Visual Studio instances:
    C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary/Build\vcvarsall.bat
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat
```

### Additional context

See analysis on https://github.com/microsoft/vcpkg/issues/22074#issuecomment-1630553428

## Expectation

vcpkg studies all available installations of Visual Studio as candidates, and not solely the first one.

Also, the text output gives the impression that multiple instances were considered with equal depth of investigation.

**Comment by LilyWangLL** [comment]
Closed as duplicate of #22074, let us continue to use #22074 to track this issue.

**Comment by Gene66-coder** [comment]
Thank you i just included on Visual Studio Build Tools the individual component "MSVC v140 - VS 2015 C++ build tools (v14.00)"
